src/Sylius/Component/Core/Model/OrderItemUnit.php

```diff
@@ -0,0 +1,200 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Component\Core\Model;
+
+use Sylius\Component\Inventory\Model\InventoryUnitInterface;
+use Sylius\Component\Inventory\Model\StockableInterface;
+use Sylius\Component\Order\Model\OrderItemUnit as BaseOrderItemUnit;
+use Sylius\Component\Shipping\Model\ShipmentInterface as BaseShipmentInterface;
+use Sylius\Component\Shipping\Model\ShippableInterface;
+
+/**
+ * @author Mateusz Zalewski <mateusz.zalewski@lakion.com>
+ */
+class OrderItemUnit extends BaseOrderItemUnit implements OrderItemUnitInterface
+{
+    /**
+     * @var string InventoryUnitInterface::STATE_*
+     */
+    protected $inventoryState = InventoryUnitInterface::STATE_CHECKOUT;
+
+    /**
+     * @var BaseShipmentInterface
+     */
+    protected $shipment;
+
+    /**
+     * @var string BaseShipmentInterface::STATE_*
+     */
+    protected $shippingState = BaseShipmentInterface::STATE_CHECKOUT;
+
+    /**
+     * @var \DateTime
+     */
+    protected $createdAt;
+
+    /**
+     * @var \DateTime
+     */
+    protected $updatedAt;
+
+    public function __construct(OrderItemInterface $orderItem)
+    {
+        parent::__construct($orderItem);
+
+        $this->createdAt = new \DateTime();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getStockable()
+    {
+        return $this->orderItem->getVariant();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setStockable(StockableInterface $stockable)
+    {
+        $this->orderItem->setVariant($stockable);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getSku()
```

Comment:
All these unnecessary methods and service will be cleared in separate PR, I suppose ;)